ufl/mathfunctions.py

```diff
@@ -296,7 +339,13 @@ def evaluate(self, x, mapping, component, index_values):
         erf = _find_erf()
         if erf is None:
             error("No python implementation of erf available on this system, cannot evaluate. Upgrade python or install scipy.")
-        return erf(a)
+        try:
+            return erf(a)
+        except TypeError:
+            if erf is scipy.special.erf:
+                return erf(a)
```

Comment:
This is dead code I think.  Either `erf(a)` succeeds (because it points at `scipy.special.erf`) or it doesn't (because `math.erf` raises TypeError), when catching the exception we can never reach this branch.